api/src/main/java/org/openmrs/module/initializer/api/c/MappingsConceptLineProcessor.java

```diff
@@ -4,36 +4,75 @@
 import org.apache.commons.lang3.StringUtils;
 import org.openmrs.Concept;
 import org.openmrs.ConceptMap;
+import org.openmrs.ConceptMapType;
 import org.openmrs.api.ConceptService;
+import org.openmrs.module.initializer.api.BaseLineProcessor;
 import org.openmrs.module.initializer.api.CsvLine;
-import org.openmrs.module.initializer.api.utils.ConceptMapListParser;
+import org.openmrs.module.initializer.api.utils.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Component;
 
 @Component("initializer.mappingsConceptLineProcessor")
 public class MappingsConceptLineProcessor extends ConceptLineProcessor {
 	
-	protected static String HEADER_MAPPINGS_SAMEAS = "same as mappings";
+	protected final Logger log = LoggerFactory.getLogger(getClass());
 	
-	private ConceptMapListParser listParser;
+	public static final String MAPPING_HEADER_PREFIX = "mapping";
+	
+	public static final String MAPPING_HEADER_SEPARATOR = "|";
+	
+	public static final String MAPPING_HEADER_SEPARATOR_REGEX = "\\|";
+	
+	public static final String HEADER_MAPPINGS_SAMEAS = "same as mappings";
 	
 	@Autowired
-	public MappingsConceptLineProcessor(@Qualifier("conceptService") ConceptService conceptService,
-	    ConceptMapListParser listParser) {
+	public MappingsConceptLineProcessor(@Qualifier("conceptService") ConceptService conceptService) {
 		super(conceptService);
-		this.listParser = listParser;
 	}
 	
 	public Concept fill(Concept concept, CsvLine line) throws IllegalArgumentException {
 		
 		if (!CollectionUtils.isEmpty(concept.getConceptMappings())) {
 			concept.getConceptMappings().clear();
 		}
-		String mappingsStr = line.get(HEADER_MAPPINGS_SAMEAS);
-		if (!StringUtils.isEmpty(mappingsStr)) {
-			for (ConceptMap mapping : listParser.parseList(mappingsStr)) {
-				concept.addConceptMapping(mapping);
+		
+		for (String header : line.getHeaderLine()) {
+			
+			String lineValue = line.get(header);
+			if (!StringUtils.isEmpty(lineValue)) {
+				
+				// For backwards-compatibility, support the original same as mappings column header
+				if (header.trim().equalsIgnoreCase(HEADER_MAPPINGS_SAMEAS)) {
+					header = MAPPING_HEADER_PREFIX + MAPPING_HEADER_SEPARATOR + ConceptMapType.SAME_AS_MAP_TYPE_UUID;
+				}
+				
+				String[] headerComponents = header.split(MAPPING_HEADER_SEPARATOR_REGEX);
```

Comment:
I think this is slightly preferable...

```suggestion
				String[] headerComponents = header.split(MAPPING_HEADER_SEPARATOR_REGEX, 3);
```

Reply:
I had this originally, but changed it to be less rigid.  There are actually up to 4 components (mappings|type|source|unique-key), though only 2 components are required.

Reply:
@ibacher I have pushed a new commit that limits the split to 4 components.